Create release preview documentation (#1425)
astronomer/docs: astro/release-previews.md

---

> sidebar_label: 'Astro release previews'

Author: ```suggestion
sidebar_label: 'Release previews'
```

> title: 'Release previews'

Author: ```suggestion
title: 'Astro release previews'
```

id: release-previews

> description: Learn how Astronomer delivers new features to Astro customers before their official release. 

Reviewer: I would just keep the description simple to: "Learn about Astronomer release previews."

---

> Astronomer shares release previews to let customers provide feedback on features before they are generally available (GA). Release previews have a different degree of feature maturity compared to GA releases. Use this document to understand release preview stages, including their different levels of availability, support, and customer access. 

Reviewer: I would prefer to keep the wording in the notion doc (https://www.notion.so/astronomerio/External-Astronomer-Release-Previews-b13c956cc6024a8ab929b515c4b16043).

"Astronomer provides previews to give customers a chance to evaluate and provide feedback on features before they are generally available (GA Release). Release previews mean a different degree of feature maturity. This article defines those release preview stages, and the levels of availability, support, and customer access granted at each stage."

Author: Can we split the diff? Just for copy editing purposes, I'd like to keep:

- "Shares release previews" over "provides previews"  (conquest against alliteration)
- "have" over "mean" in the second sentence.
- "document" over "article"

To view support promises for GA releases of Runtime, see [Astro Runtime maintenance and lifecycle policy](runtime-version-lifecycle-policy.md). To submit feedback or a support request for a Public or Private Preview feature, see [Submit a support request](astro-support.md).

## Astro release previews

Astro features can be shared as Private Preview or Public Preview. The table below outlines how these release previews differ:

> | Release Preview Type | Customer Access | Availability                  | Support                                                                                                                                                                                                                                                   |

Author: @juliana-astro I thought that the table was able to capture all of the information you included about these release types, but let me know if I missed anything.

Reviewer: The entire notion doc should be covered in the web page. I updated the wording above the table:
"Today Astro features can be previewed in a Private Preview or Public Preview. The table below offers a snapshot on how the release previews differ:"

| -------------------- | --------------- | ----------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Private Preview      | Invite only.    | No SLA or SLO.                | Product and R&D will be in active communication with customers who are have a Private Preview of a feature to receive feedback, bug reports, questions, and more. Customer Reliability Engineering (CRE) does not provide support for features in Private Preview.                                                                                             |

Reviewer: ```suggestion
| Private Preview      | Invite only.    | No SLA or SLO.                | Product and R&D will be in active communication with customers who are in a Private Preview of a feature to receive feedback, bug reports, questions, and more. Customer Reliability Engineering (CRE) does not provide support for features in Private Preview.                                                                                             |
```

> | Public Preview       | Everyone.       | No SLA. Internal SLOs supported by R&D and CRE. | CRE treats all issues within the feature as P3 or P4, depending on the nature of the issue. CRE continues to support remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview. |

Author: @juliana-astro Where would the Runtime executor fall on this list? It needs to be flipped on manually but it's available to all customers.

Reviewer: Given customers can enable this themselves, I'd say it falls under Public Preview.

> | Public Preview       | Everyone.       | No SLA. Internal SLOs supported by R&D and CRE. | CRE treats all issues within the feature as P3 or P4, depending on the nature of the issue. CRE continues to support remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview. |

Reviewer: Add below the table this section on more detail -- 

## Private Preview

Private Preview offers a select group of customers early access to a feature— to test, offer feedback, and preview a feature before it becomes more widely available.

CUSTOMER ACCESS

- An invite to a Private Preview will come from your Customer Success Manager or a member of our Product Team.

AVAILBILITY

- No SLA is set for a feature in Private Preview. Other availability metrics may be set, such as an internal SLO.

SUPPORT

- Product and R&D will be in active communication with customers who are in Private Preview of a feature to hear feedback and provide support: including observations on the experience, triaging bugs, and answering questions.
- Our Customer Reliability Engineering (CRE) team does not provide support for features in Private Preview.

## Public Preview

Public Preview offers all customers access to a feature before it is released for GA. The key differentiator between Public Preview and General Availability is in availability and support for the feature. 

CUSTOMER ACCESS

- All Astro customer orgs have access to the feature. Note: some features may have access gated by Role-Based Access Control.

AVAILBILITY

- A feature in Public Preview does not have the same SLA commitment as a GA feature.
- Internal Service Level Objectives (SLO’s) will be set to monitor and support the feature’s availability.

SUPPORT

- CRE will treat all issues within a Public Preview feature as P3 or P4, depending on the nature of the issue:
    - P3: Loss of use of the feature (not affecting service in general). Feature not working as expected.
    - P4: Inquiry about the feature. Information requested on capabilities, navigation, installation, or configuration. Bug affecting a small number of users.
- CRE continues to support issue remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview.

Author: Cool with publishing this content for now, but in the future I think it would be beneficial if we could sort everything into a table like Databricks does here for clarity: https://docs.databricks.com/release-notes/release-types.html#runtime-releases

---
sidebar_label: 'Astro release previews'
title: 'Release previews'
id: release-previews
description: Learn about Astronomer release previews. 
---

> Astronomer shares release previews to give customers a chance to evaluate and provide feedback on features before they are generally available (GA). Release previews have a different degree of feature maturity. This document defines those release preview stages, and the levels of availability, support, and customer access granted at each stage.

Reviewer 2: ```suggestion
Astronomer release previews provide customers with an opportunity to evaluate and provide feedback on features before they are generally available (GA). This document defines the different release preview types, their availability, the support provided for each release preview type, and how and when they can be accessed.
```

> To view support promises for GA releases of Runtime, see [Astro Runtime maintenance and lifecycle policy](runtime-version-lifecycle-policy.md). To submit feedback or a support request for a Public or Private Preview feature, see [Submit a support request](astro-support.md).

Reviewer 2: ```suggestion
To view the support commitment for GA releases of Runtime, see [Astro Runtime maintenance and lifecycle policy](runtime-version-lifecycle-policy.md). To submit feedback or a support request for a Public or Private Preview feature, see [Submit a support request](astro-support.md).
```

## Astro release previews

> Astro features can be shared as Private Preview or Public Preview. The table below offers a snapshot on how the release previews differ:

Reviewer 2: ```suggestion
Astro features are shared as Private Preview or Public Preview. The following table describes the differences between the different release preview types.
```

![image](https://user-images.githubusercontent.com/104205257/199968348-50698d6c-6023-4436-a3cb-393306493a2b.png)

> | Release Preview Type | Customer access | Availability                                    | Support                                                                                                                                                                                                                                                          |

Reviewer 2: ```suggestion
| Release Preview Type | Customer Access | Availability                                    | Support                                                                                                                                                                                                                                                          |
```

| -------------------- | --------------- | ----------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Private Preview      | Invite only.    | No SLA or SLO.                                  | Product and R&D will be in active communication with customers who are in a Private Preview of a feature to receive feedback, bug reports, questions, and more. Customer Reliability Engineering (CRE) does not provide support for features in Private Preview. |

Reviewer 2: ```suggestion
| Private Preview      | Invite only.    | No SLA or SLO                                  | Product and Research and Development  actively communicate with customers who are using a Private Preview of a feature to receive feedback, bug reports, questions, and more. Customer Reliability Engineering (CRE) does not provide support for features in Private Preview. |
```
Have SLA and SLO been previously defined? If not, they should be defined here.

Reviewer 2: ![image](https://user-images.githubusercontent.com/104205257/199988062-731b69e3-1ed3-4939-bc62-c298b8f8a2a5.png)

> | Public Preview       | Everyone        | No SLA. Internal SLOs supported by R&D and CRE. | CRE treats all issues within the feature as P3 or P4, depending on the nature of the issue. CRE continues to support remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview.        |

Reviewer 2: ```suggestion
| Public Preview       | Everyone        | No SLA. Internal SLOs supported by Research and Development and CRE | The severity of the issue determines how CRE responds to issues with Public Preview features. Issues can be assigned a P3 or P4 priority level. CRE uses defined priority levels to resolve of all other issues, including platform issues caused by features in Public Preview.        |
```
Have the P3 or P4 priority levels been defined elsewhere, or is there a document that provides this information? Definitions for each priority level should be provided.

> | Public Preview       | Everyone        | No SLA. Internal SLOs supported by R&D and CRE | CRE treats all issues related to the feature at a P3 or P4 priority level. See [Submit a support request](https://docs.astronomer.io/astro/astro-support#best-practices-for-support-request-submissions). CRE continues to support remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview.       |

Reviewer 2: ```suggestion
| Public Preview       | Everyone        | No SLA. Internal SLOs supported by Research and Development and CRE | CRE treats all issues related to the feature at a P3 or P4 priority level. See [Submit a support request](https://docs.astronomer.io/astro/astro-support#best-practices-for-support-request-submissions). CRE continues to support remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview.       |
```

## Private Preview

Private Preview offers a select group of customers early access to a feature— to test, offer feedback, and preview a feature before it becomes more widely available.

### Customer access 

> An invite to a Private Preview will come from your Customer Success Manager or a member of the Astronomer Product team.

Reviewer 2: ```suggestion
Your organization is invited to participate in a Private Preview by your Customer Success Manager or a member of the Astronomer Product team.
```

### Availability 

> No SLA is set for a feature in Private Preview. Other availability metrics may be set, such as an internal SLO.

Reviewer 2: ```suggestion
SLAs are not available for Private Preview features. Other agreements such as an internal SLO might be provided at the discretion of Astronomer.
```

### Support

> Product and R&D will be in active communication with customers who are in Private Preview of a feature to hear feedback and provide support, including observations on the experience, triaging bugs, and answering questions.

Reviewer 2: ```suggestion
Product and Research and Development actively communicate with customers who are using a feature in Private Preview to solicit feedback and provide support. This includes observations on their experience, triaging bugs, and answering questions.
```

> Astronomer's Customer Reliability Engineering (CRE) team does not provide support for features in Private Preview.

Reviewer 2: ```suggestion
The CRE team doesn't support Private Preview features.
```
CRE has been defined previously. It did not include the modifier _Astronomer_.

## Public Preview

> Public Preview offers all customers access to a feature before it is released for GA. The key difference between Public Preview and General Availability is in availability and support for the feature. 

Reviewer 2: ```suggestion
All customers can access Public Preview features before they become GA. The primary difference between Public Preview and GA is the availability of the feature and the support provided by Astronomer. 
```

### Customer access 

> All Astro customer orgs have access to the feature. Note: some features may have access gated by Role-Based Access Control.

Reviewer 2: ```suggestion
All Astro customers can access the feature. Some feature access might be restricted by role-based access control (RBAC).
```

### Availability 

> A feature in Public Preview does not have the same SLA commitment as a GA feature. Internal service level objectives (SLO’s) will be set to monitor and support the feature’s availability.

Reviewer 2: ```suggestion
Public Preview features don't have the same SLA commitment as GA features. Astronomer sets Internal SLOs to monitor and support Public Preview features.
```

### Support

> CRE will treat all issues within a Public Preview feature as P3 or P4, depending on the nature of the issue:

Reviewer 2: ```suggestion
CRE assigns a P3 or P4 severity level to all Public Preview issues. The following are the definitions for each severity level:
```

> - P3: Loss of use of the feature (not affecting service in general). Feature not working as expected.

Reviewer 2: ```suggestion
- P3: Loss of use of a feature, or a feature is not working as expected. Service is not affected..
```

> - P4: Inquiry about the feature. Information requested on capabilities, navigation, installation, or configuration. Bug affecting a small number of users.

Reviewer 2: ```suggestion
- P4: Feature inquiries, or bugs limited to a small number users. Inquiries can include questions about feature capabilities, navigation, installation, or configuration.
```

> CRE continues to support issue remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview.

Reviewer 2: ```suggestion
CRE uses defined priority levels to resolve of all other issues, including platform issues caused by features in Public Preview.
```

Reviewer 2: I'm not sure what value this statement adds.

Author: I think the gist of this statement is "If a Public Preview feature causes an issue with the general platform, that issue will be treated with the priority of a general release". We could be more explicit there

> CRE continues to support issue remediation for all other issues at defined priority levels, including if a platform issue is caused by a feature in Public Preview.

Author: ```suggestion
If a feature in Public Preview causes an issue with the data plane, CRE treats that issue with the priority of a generally available feature. 
```
